findOne throws exception instead of returning null when resource not found

`findOne` should return `null` when no document is found with the given `id`. https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/repository/CrudRepository.html#findOne-ID-

Instead it is throwing an exception:

```
java.lang.RuntimeException: findById exception
	at com.microsoft.azure.spring.data.documentdb.core.DocumentDbTemplate.findById(DocumentDbTemplate.java:117)
	at com.microsoft.azure.spring.data.documentdb.repository.support.SimpleDocumentDbRepository.findOne(SimpleDocumentDbRepository.java:64)
⋮
Caused by: com.microsoft.azure.documentdb.DocumentClientException: Message: {"Errors":["Resource Not Found"]}
```

--- a/documentdb/spring-data-azure-documentdb/src/main/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplate.java
+++ b/documentdb/spring-data-azure-documentdb/src/main/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplate.java
@@ -16,6 +16,7 @@ import com.microsoft.azure.documentdb.Resource;
 import com.microsoft.azure.documentdb.SqlParameter;
 import com.microsoft.azure.documentdb.SqlParameterCollection;
 import com.microsoft.azure.documentdb.SqlQuerySpec;
+import com.microsoft.azure.documentdb.internal.HttpConstants;
 import com.microsoft.azure.spring.data.documentdb.DocumentDbFactory;
 import com.microsoft.azure.spring.data.documentdb.core.convert.MappingDocumentDbConverter;
 import com.microsoft.azure.spring.data.documentdb.core.mapping.DocumentDbPersistentEntity;
@@ -114,6 +115,10 @@ public class DocumentDbTemplate implements DocumentDbOperations, ApplicationCont
                 return null;
             }
         } catch (DocumentClientException e) {
+            if (e.getStatusCode() == HttpConstants.StatusCodes.NOTFOUND){
+                return null;
+            }
+
             throw new RuntimeException("findById exception", e);
         }
     }

--- a/documentdb/spring-data-azure-documentdb/src/main/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplate.java
+++ b/documentdb/spring-data-azure-documentdb/src/main/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplate.java
@@ -115,7 +115,7 @@ public class DocumentDbTemplate implements DocumentDbOperations, ApplicationCont
                 return null;
             }
         } catch (DocumentClientException e) {
-            if (e.getStatusCode() == HttpConstants.StatusCodes.NOTFOUND){
+            if (e.getStatusCode() == HttpConstants.StatusCodes.NOTFOUND) {
                 return null;
             }
 

--- a/documentdb/spring-data-azure-documentdb/src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java
+++ b/documentdb/spring-data-azure-documentdb/src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java
@@ -27,7 +27,6 @@ import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 import java.util.List;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.assertj.core.api.Assertions.filter;
 
 @RunWith(SpringJUnit4ClassRunner.class)
 @PropertySource(value = {"classpath:application.properties"})

--- a/documentdb/spring-data-azure-documentdb/src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java
+++ b/documentdb/spring-data-azure-documentdb/src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java
@@ -27,13 +27,17 @@ import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 import java.util.List;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.filter;
 
 @RunWith(SpringJUnit4ClassRunner.class)
 @PropertySource(value = {"classpath:application.properties"})
 public class DocumentDbTemplateIT {
 
+    private static final String TEST_ID = "testid";
+    private static final String TEST_NOTEXIST_ID = "testid2";
+
     private static final String TEST_DB_NAME = "testdb";
-    private static final Person TEST_PERSON = new Person("testid", "testfirstname", "testlastname");
+    private static final Person TEST_PERSON = new Person(TEST_ID, "testfirstname", "testlastname");
 
     @Value("${documentdb.uri}")
     private String documentDbUri;
@@ -93,6 +97,9 @@ public class DocumentDbTemplateIT {
         assertThat(result.getId()).isEqualTo(TEST_PERSON.getId());
         assertThat(result.getFirstName()).isEqualTo(TEST_PERSON.getFirstName());
         assertThat(result.getLastName()).isEqualTo(TEST_PERSON.getLastName());
+
+        final Person nullResult = dbTemplate.findById(Person.class.getSimpleName(), TEST_NOTEXIST_ID, Person.class);
+        assertThat(nullResult).isNull();
     }
 
     @Test

